packages/core/admin/admin/src/content-manager/history/components/VersionHeader.tsx

```diff
@@ -18,9 +20,24 @@ export const VersionHeader = ({ headerId }: VersionHeaderProps) => {
     version: state.selectedVersion,
     layout: state.layout,
   }));
+  const [{ query }] = useQueryParams<{
+    plugins?: Record<string, unknown>;
+  }>();
 
   const mainFieldValue = version.data[layout.contentType.settings.mainField];
 
+  const backLink = (() => {
```

Comment:
I'm not sure about this, feels a bit weird to see an iife just to solve the backlink, why are you not just using a memo? I guess this is a memo but with the memoisation.

Your route should have `:collectionType`, why not just use that instead?

```ts
const params = useParams<{collectionType: string; slug: string}>()

const backLink = {
	to: `../${collectionType}/${slug}/${version.relatedDocumentId}`
	search: stringify({ plugins: query.plugins });
}
```

Reply:
I'm not concerned about performance here, I don't think a memo is necessary.

For context, I'm using a function because I like the syntax it offers, with if blocks and early returns. I find it more readable than nested ternaries or nested template strings. But writing a function only meant to compute one value in one place felt silly, so I'd self-invoke it to "bypass" the function. I changed it to be a regular function though, if that makes it clearer.

Also good point about the :collectionType, I forgot I could get it from there :)